lib/tokenize.cpp

```diff
@@ -2003,9 +2003,19 @@ bool Tokenizer::simplifyUsing()
         // Move struct defined in using out of using.
         // using T = struct t { }; => struct t { }; using T = struct t;
         // fixme: this doesn't handle attributes
-        if (Token::Match(start, "struct|union|enum %name%| {")) {
-            if (start->strAt(1) != "{") {
-                Token *structEnd = start->linkAt(2);
+        if (Token::Match(start, "class|struct|union|enum %name%| {|:")) {
+            if (Token::Match(start->next(), "%name%")) {
+                Token *structEnd;
+                if (start->strAt(2) == "{")
+                    structEnd = start->linkAt(2);
+                else {
+                    structEnd = start->tokAt(3);
```

Comment:
code could be `struct { int x; } s;` right?

Reply:
We are skipping over the base class information starting after `:` and continuing until `{` is found where we jump over the contents of the struct by using the link.  We use `structEnd' directly as a pointer we increment rather than a temp pointer which we copy to `structEnd` when we get to the end.

I just noticed we don't handle the case where we have a unnamed derived struct.